### PR TITLE
Add diff command

### PR DESCRIPTION
**Pull request by yan-zaretskiy**
`eclair diff` will dump all common keywords for a pair of summary files to an HTML report. The actual difference in data is not calculated yet.

### CODE REVIEW

File: eclair/src/diff.rs [hunk 1]
```diff
@@ -1,17 +1,120 @@
...
+use std::fs::File;
+use std::io::prelude::*;
 
+use askama::Template;
 use chrono::Duration;
 use chrono::{NaiveDate, NaiveDateTime, NaiveTime};
-use plotly::common::{Mode, Title};
-use plotly::layout::{Axis, Layout};
-use plotly::{Plot, Scatter};
+use itertools::EitherOrBoth;
```
Comment (ErichDonGubler) [inline eclair/src/diff.rs:8]: Might I suggest deciding on a normalized import style? :)

File: eclair/src/diff.rs [hunk 2]
```diff
@@ -1,17 +1,120 @@
...
+
+fn merge(t1: &[i64], t2: &[i64]) -> Merged {
+    t1.iter()
+        .copied()
+        .enumerate()
+        .merge_join_by(t2.iter().copied().enumerate(), |i, j| i.1.cmp(&j.1))
+        .collect()
+}
+
+type Padded = Vec<[Option<f32>; 2]>;
```
Comment (ErichDonGubler) [inline eclair/src/diff.rs:25]: You could easily use an `ArrayVec` here and avoid allocation penalties.

File: eclair/src/diff.rs [hunk 3]
```diff
@@ -43,32 +146,27 @@ fn extract_time_and_data(summary: &Summary) -> (Vec<String>, HashMap<&ItemId, us
     (times, data_item_map)
 }
 
-pub fn diff(candidate: &Summary, reference: &Summary) {
+pub fn diff(candidate: &Summary, reference: &Summary, output: Option<PathBuf>) {
```
Comment (ErichDonGubler) [inline eclair/src/diff.rs:149]: Look like `output` only need be a `Option<&Path>` -- client code can allocate if they need to.

File: eclair/src/diff.rs [hunk 4]
```diff
@@ -80,14 +178,21 @@ pub fn diff(candidate: &Summary, reference: &Summary) {
...
+        common_plots,
+    };
+
+    let rendered = diff_html.render().unwrap();
+
+    let mut file = if let Some(name) = output {
+        File::create(name).unwrap()
+    } else {
+        File::create("diff.html").unwrap()
+    };
```
Comment (ErichDonGubler) [inline eclair/src/diff.rs:196]: You could push this `if` condition into the argument position if you wanted:

```suggestion
    let mut file = 
        File::create(if let Some(name) = output {
            name
        } else {
            "diff.html"
        }).unwrap();
```

File: eclair/src/diff.rs (hunk 2 above)
Comment (yan-zaretskiy) [reply eclair/src/diff.rs:25]: Could you elaborate?

Comment (ErichDonGubler) [reply eclair/src/diff.rs:25]: ~You should literally be able to `s/Vec/ArrayVec/` here and it Just Work® -- and then you save yourself a heap allocation at this level. :)­~

I'm actually not so sure in retrospect, apologies. If you have a static max length that wouldn't be outrageous for the stack, `Padded` could just live on the stack. But I think I suggested too quickly, now that I review this again.

**Pull request merged**